anndata/_core/sparse_dataset.py

```diff
@@ -292,11 +293,21 @@ class BaseCompressedSparseDataset(ABC):
 
     def __init__(self, group: h5py.Group | ZarrGroup):
         type(self)._check_group_format(group)
-        self.group = group
+        self._group = group
 
     shape: tuple[int, int]
     """Shape of the matrix."""
 
+    @property
+    def group(self):
+        return self._group
```

Comment:
Yes.

Reply:
So the type hint appears to be unnecessary.  The code can infer it.  You'll notice that some of the builds failed because of documentation issues - I honestly don't know why.  The other references to `Group` have no issue, but this one causes things to break.  VSCode can infer the type and the docs are fine without it.

Reply:
Sorry I lied.  The docs don't work but the type hints do work.  I was looking at the wrong `Group`

Reply:
Yeah, I have no idea why this is breaking. Extremely strange stuff.  I will look again tomorrow.

Reply:
This looks like it's working. Can we resolve?

Reply:
Well I did not push the change, so the docs are building because of that, unless this works for you locally.